roles/cdi/tasks/deprovision.yml

```diff
@@ -7,40 +7,97 @@
 - name: Delete {{ cdi_namespace }} ResourceQuota
   command: kubectl delete -f /tmp/cdi-deprovision-resourcequota.yml -n {{ cdi_namespace }} --ignore-not-found
 
-- name: Check that cdi-controller.yaml still exists in /tmp
+- name: Check that cdi-operator-cr.yaml still exists in /tmp
   stat:
-    path: "/tmp/cdi-controller.yaml"
-  register: cdi_template
-
-- name: Get cdi-controller.yaml
-  block:
-    - name: Check for cdi-controller.yaml.j2 template in {{ cdi_template_dir }}
-      stat:
-        path: "{{ cdi_template_dir }}/cdi-controller.yaml.j2"
-      register: byo_template
-
-    - name: Render CDI deprovision yaml
-      template:
-        src: "cdi-controller.yaml.j2"
-        dest: "/tmp/cdi-controller.yaml"
-      when: byo_template.stat.exists == true
-
-    - name: Download and render cdi-controller.yaml from {{ cdi_release_manifest_url }}
-      block:
-        - name: Download CDI Template
-          get_url:
-            url: "{{ cdi_release_manifest_url }}/{{ release_tag }}/cdi-controller.yaml.j2"
-            dest: "/tmp/cdi-controller.yaml.j2"
-
-        - name: Render CDI deprovision yaml
-          template:
-            src: "/tmp/cdi-controller.yaml.j2"
-            dest: "/tmp/cdi-controller.yaml"
-
-      when: byo_template.stat.exists == false
-
-  when: cdi_template.stat.exists == false
-
-- name: Delete CDI Resources
-  command: kubectl delete -f /tmp/cdi-controller.yaml -n {{ cdi_namespace }} --ignore-not-found
+    path: "/tmp/cdi-operator-cr.yaml"
+  register: cdi-operator-cr
+
+- name: Render CDI operator cr
+  template:
+    src: "cdi-operator-cr.yaml.j2"
+    dest: "/tmp/cdi-operator-cr.yaml"
+  when: cdi-operator-cr.stat.exists == False
+
+- name: Ask the operator to cleanup CDI
+  command: "{{ cluster_command }} delete -f /tmp/cdi-operator-cr.yaml --ignore-not-found=true"
+
+- name: Wait until cdi-api is deleted
```

Comment:
with os 3.11 all these waits could be replaced with a `kubectl wait cdis.cdi.kubevirt.io/cdi --for=delete` because kubectl does a foreground delete. So all the children of the cdi object will be deleted before the root.  But I don't think that's the case with k8s 1.12.  Also, I guess it doesn't hurt to have this stuff here.